### PR TITLE
FINERACT-2205: Add tests when charge is added after the charge off date and behavior is zero-interest

### PR DESCRIPTION
**Pull request by mariiaKraievska**
## Description

Add tests for test cases S9-3 and S9-4 when behaviour is "zero-interest" and forbid backdated charge-off transactions in the middle of the monetary activity.

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java [hunk 1]
```diff
@@ -3581,4 +3581,17 @@ public boolean isChargeOffOnDate(final LocalDate onDate) {
...
     }
 
+    public boolean hasMonetaryActivityAfter(final LocalDate transactionDate) {
+        for (LoanTransaction transaction : this.getLoanTransactions()) {
+            if (transaction.getTransactionDate().isAfter(transactionDate) && !transaction.isNonMonetaryTransaction()) {
+                return true;
+            }
+        }
+        for (LoanCharge loanCharge : this.getLoanCharges()) {
+            if (!loanCharge.isPaid() && loanCharge.getSubmittedOnDate().isAfter(transactionDate)) {
```
Comment (adamsaghy) [inline fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java]: You might wanna include `isWaived` and `isWrittenOff` as well.

Comment (mariiaKraievska) [reply fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java]: Done, PTAL

**Review by adamsaghy — APPROVED** [review]
LGTM
